src/main/scala/io/github/pauljamescleary/petstore/validation/PetValidationInterpreter.scala

```diff
@@ -12,4 +12,15 @@ class PetValidationInterpreter(implicit repository: PetRepositoryAlgebra[IO]) ex
       matches.forall(possibleMatch => possibleMatch.bio != pet.bio)
     }.map(_ => ())
   }
+
+  def exists(petId: Option[Long]): IO[Unit] = {
```

Comment:
curious: any reason you decide to `raiseError` instead of using something like `Either` to bring up the fact that the pet was not found?

Reply:
Your question hints a little at how I structured validations so far.

As of right now, the idea is that validations are `IO[Unit]`, where we assume that we will get a failure inside of an `IO` if the validation fails; otherwise we get a successful result which is always `Unit`

I am most definitely open to other ways to do business validations.

I decided not to use `Validated` at this point because I am not interested in aggregating errors ala applicatives.

So, to answer your question directly, I am using `IO.raiseError` to "lift" the error into the `IO` monad

Reply:
In some further conversations, I believe that the validation effect type is incorrect.  We should be using an `EitherT[IO, BusinessError, A]` as the effect type for validations.

We need to do a natural transformation in the service for lifting

I have some works to do on this part.  Welcome to ideas as I do some codeplay.

Reply:
Sounds good to me, I must say I'm experimenting with all of this as well.
I was mostly referring to something like you suggest in your second comment. I'll let you know if I think of something worth sharing.

Thanks